From d3e71de054f5448d4766c8911350d3eaca60a0e2 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Thu, 26 May 2016 14:36:40 +0300
Subject: [PATCH 0243/1345] net: mvpp2x: prefetch cacheline of buffer data in
 mv_pp2x_rx

commit  6356732247393eb96c950ede0a7c4a4ab523318e from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

-gives better performance during tested throughput benchmarks

Change-Id: Iedfb4f0f3f80df096e0cf2108e3f9526b08cd8ac
Reviewed-on: http://vgitil04.il.marvell.com:8080/30022
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index 54b7921..ea8129d 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -2202,7 +2202,8 @@ static int mv_pp2x_rx(struct mv_pp2x_port *port, struct napi_struct *napi,
 #ifdef MVPP2_VERBOSE
 		mv_pp2x_skb_dump(skb, rx_desc->data_size, 4);
 #endif
-
+		/* Prefetch 128B packet_header */
+		prefetch(data + NET_SKB_PAD);
 		dma_sync_single_for_cpu(dev->dev.parent, buf_phys_addr,
 					MVPP2_RX_BUF_SIZE(rx_desc->data_size),
 					DMA_FROM_DEVICE);
-- 
1.7.9.5

